index.bs

```diff
@@ -1710,3 +1738,161 @@ When [=a browsing context is discarded=], run the following steps:
  1. [=Emit an event=] with |body| and |parent id|.
 
 </div>
+
+## Log ## {#module-log}
+
+The <dfn export for=modules>log</dfn> module contains functionality and events
+related to logging.
+
+### Definition ### {#module-log-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+Command = (GetCommand)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+LogEvent = (
+  LogEntryAddedEvent
+)
+</pre>
+
+### Types ### {#module-log-types}
+
+#### Log Entry #### {#types-log-log-entry}
+
+<pre class="cddl local-cddl">
+
+LogLevel = "debug" / "info" / "warning" / "error"
+
+LogEntry = {
+  type: text,
+  level: LogLevel,
+  text: text / null,
+  timestamp: int,
+  extra: { *text => any }
+}
+
+ConsoleLogExtra = {
+  type: text,
+  realm: Realm,
+  args: [RemoteValue*],
+}
+
+JavascriptLogExtra = {
+  lineNumber: int,
+  columnNumber: int,
+  url: text
+}
+
+</pre>
+
+Each log event is represented by a <code>LogEntry</code> object. This has a
+<code>kind</code> property which represents the kind of log entry added, a
+<code>level</code> property representing serverity, a <code>text</code> property
+with the log message string itself, and a <code>timesteamp</code> property
+corresponding to the time the log entry was generated. The <code>extra</code>
+property contains additional data specific to the given type.
```

Comment:
I wonder if we could describe the properties as part of the CDDL definition instead of spec text, e.g.

```
LogEntry = {
  type: text,             # represents the kind of log entry added
  level: LogLevel,        # log severity
  text: text / null,      # log message
  timestamp: int,         # time the log entry was generated
  extra: { *text => any } # additional data specific to the given type
}
```

This could be useful later on for code generation.

Reply:
Yeah, we definitely could do that, but it would be quite different from the way that IDL fragments work where all the non-normative descriptive stuff is in the text. I'd be interested in other opinions here.

Reply:
I'm going to resolve this on the basis the comment isn't going to block this PR, but please feel free to open an Issue with this suggestion.